releasetool/commands/start/go.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import getpass
+import json
+import os
+import subprocess
+import textwrap
+from typing import Optional, Dict, Sequence
+
+import attr
+import click
+
+import releasetool.filehelpers
+import releasetool.git
+import releasetool.github
+import releasetool.secrets
+import releasetool.commands.common
+
+
+_CHANGELOG_FILENAME = "CHANGES.md"
+
+_CHANGELOG_TEMPLATE = """\
+# Changes
+
+"""
+
+
+@attr.s(auto_attribs=True, slots=True)
+class Context:
+    module_name: str = None
+    relative_module_name: str = None  # relative to repo root; used for tags
+    changes: Sequence[str] = ()
+    release_notes: Optional[str] = None
+    last_release_version: Optional[str] = None
+    last_release_committish: Optional[str] = None
+    release_version: Optional[str] = None
+    release_branch: Optional[str] = None
+#    pull_request: Optional[dict] = None
+
+
+def determine_module_name(ctx: Context) -> None:
+    # Get the module name from the go.mod file in the current directory.
+    click.secho("> Figuring out the module name.", fg="cyan")
+    info = read_gomod()
+    if info is None:
+        click.secho("Looks like we're releasing the repo (no modules).")
+    else:
+        ctx.module_name = info["Module"]["Path"]
+        ctx.relative_module_name = relative_module_name(ctx.module_name)
+        click.secho(f"Looks like we're releasing {ctx.module_name} (relative path {ctx.relative_module_name}).")
+
+def read_gomod() -> Optional[Dict[str, str]]:
+    if os.path.isfile("go.mod"):
+        output = subprocess.check_output(["go", "mod", "edit", "-json"]).decode("utf-8")
+        return json.loads(output)
+    elif os.path.isdir(".git"):
+        # No go.mod; must release from repo root.
+        return None
+
+# Returns modname relative to the repo root, assuming modname's go.mod file is
```

Comment:
You can use a [docstring](https://www.python.org/dev/peps/pep-0257/) for this.

Reply:
I tried all the following ways to install nox, but in every case `nox blacken` resulted in `bash: nox: command not found`.

```
pip3 install nox 
pip3 install nox-automation
pip3 install --user --upgrade nox
pip install --user --upgrade nox
```

Reply:
Docstring done.

Reply:
@jba Could you try invoking nox [via the interpreter](https://nox.readthedocs.io/en/stable/usage.html#command-line-usage)? `python3 -m nox blacken`

Reply:
Successfully ran `nox blacken`.